### ISSUE TITLE
Switching pane-maximize will add an additional line break in the new window

### ISSUE THREAD

**Issue by dhslegen** [Eugeny/tabby#8071]
**Describe the problem**:
Switching pane-maximize will add an additional line break in the new window, which can be solved by changing a shortcut key, but I'm still used to the combination of cmd option enter.

<img width="566" alt="image" src="https://user-images.githubusercontent.com/37139487/224586514-f71b8f60-7264-4df5-9c87-4cbd7fab41b4.png">

**To Reproduce**:
split window>cmd option enter * n

**Comment by Eugeny** [comment]
of course!

**Comment by echo304** [comment]
@Eugeny Can you provide some advice that where all the key strokes are handled?
I suspected that predefined hotkey(cmd + option + enter) somehow affect to the terminal command line.
Reason why I think is that if I change hotkey to other key combination without 'enter' it does not add line break.

**Comment by Eugeny** [comment]
All hotkeys are handled in https://github.com/Eugeny/tabby/blob/master/tabby-core/src/services/hotkeys.service.ts , the likely solution is to `preventDefault` the keyboard event once a hotkey is recognized

**Comment by echo304** [comment]
I tried to figure out where those keyboard event is propagated in `hotkeys.service.ts` but it didn't work well.
I guess there is leaking(propagated) keyboard event of "Enter" somewhere... 🕵️ 
So, I'm going to dig where actual input is added to the UI. https://github.com/Eugeny/tabby/blob/fac6ec572f3f173a327b2414f1f5ab5848cd8fe2/tabby-terminal/src/api/baseTerminalTab.component.ts#L439-L447
Is this right place to start? 🤔 
@Eugeny

**Comment by Eugeny** [comment]
Not exactly, xterm is listening to keyboard events directly, so terminal input doesn't to through tabby code, but I'm intercepting all keys in HotkeyService first, where, if a key matches a known shortcut, it should be cancelled and never make it's way into xterm

**Comment by echo304** [comment]
Got it. 👍

**Comment by echo304** [comment]
If I commented out these lines, this bug is removed but simply removing these lines is definitely not the solution.
I briefly understand why they are necessary thanks to the comment `// macOS will swallow non-modified keyups if Cmd is held down`. Can you explain the more detailed background of this code snippet? 

https://github.com/Eugeny/tabby/blob/fac6ec572f3f173a327b2414f1f5ab5848cd8fe2/tabby-core/src/services/hotkeys.service.ts#L181-L184

**Comment by Eugeny** [comment]
Taking these lines out won't do anything for other operating systems unfortunately, as macOS is the only one with this problem. On other OSes, `keyup` will still fire by itself as expected. The problem is that HotkeyService doesn't seem to intercept/cancel `keyup` that corresponds to a recognized shortcut, and it's getting passed down to the terminal.
